### PR TITLE
feat: kvselect component was rewritten and basic test file was added

### PR DESCRIPTION
**Pull request by roger-in-kiva**
kvselect component was rewritten to use composition api and a basic test file was added

### CODE REVIEW

**Review by emuvente — COMMENTED** [review]
I think this just needs the `emits: ['change', 'update:modelValue']` option defined, and then this should be good to go!

**Comment by roger-in-kiva** [comment]
> I think this just needs the `emits: ['change', 'update:modelValue']` option defined, and then this should be good to go!

Fixed, thanks @emuvente !

File: @kiva/kv-components/vue/KvSelect.vue [hunk 1]
```diff
@@ -6,6 +6,7 @@
 				:id="id"
 				:disabled="disabled"
 				:value="value"
+				role="select"
```
Comment (ryan-ludwig) [inline @kiva/kv-components/vue/KvSelect.vue:9]: This role can be removed as it is redundant on a `<select>` element

Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvSelect.vue]: Nice point, thanks. Is fixed.

File: @kiva/kv-components/vue/KvSelect.vue [hunk 2]
```diff
@@ -58,21 +58,32 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+		/**
```
Comment (ryan-ludwig) [inline @kiva/kv-components/vue/KvSelect.vue:61]: We didn't need this extra prop in the vue2 version. I think we can just use the value prop as before.

File: @kiva/kv-components/vue/KvSelect.vue [hunk 3]
```diff
@@ -58,21 +58,32 @@ export default {
...
-	data() {
-		return {
-			mdiChevronDown,
-		};
-	},
-	methods: {
-		onChange(event) {
+	emits: [
+		'change',
+		'update:modelValue',
```
Comment (ryan-ludwig) [inline @kiva/kv-components/vue/KvSelect.vue:71]: this would be `'update:value'`

File: @kiva/kv-components/vue/KvSelect.vue [hunk 4]
```diff
@@ -58,21 +58,32 @@ export default {
...
+		const onChange = (event) => {
 			/**
 			* The value that the select has changed to
 			* @event change
 			* @type {Event}
 			*/
-			this.$emit('change', event.target.value);
-		},
+			emit('change', event.target.value);
+			emit('update:modelValue', event.target.value);
```
Comment (ryan-ludwig) [inline @kiva/kv-components/vue/KvSelect.vue:81]: and `emit('update:value', event.target.value);`

File: @kiva/kv-components/vue/KvSelect.vue (hunk 3 above)
Comment (ryan-ludwig) [reply @kiva/kv-components/vue/KvSelect.vue:71]: Ahh, thanks for the link. In that case I think we should remove the `value` prop

Comment (emuvente) [reply @kiva/kv-components/vue/KvSelect.vue:71]: Agreed, the `value` prop should be removed and `modelValue` should replace any usage of it. That will also mean that the model option should changed to:
```
model: {
	prop: 'modelValue',
	event: 'update:modelValue',
},
```

Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvSelect.vue:64]: Good!
The prop was removed and replaced with modelValue.

**Comment by roger-in-kiva** [comment]
Ready for testing!

**Review by emuvente — APPROVED** [review]
Looks great!

**Pull request merged**